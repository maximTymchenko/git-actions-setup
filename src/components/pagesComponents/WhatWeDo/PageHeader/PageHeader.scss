@import '../../../../styles/vars';

.page-header {
    width: 100%;
    max-width: $container__size - 100px;
    //   max-width: $header__width;
    margin: 210px auto 0;
    @include main__padding;
    position: relative;
    left: 30px;

    &__wrapper {
        margin-top: 70px;
    }

    &__container {
        display: flex;
        justify-content: space-between;
    }

    &__title {
        text-align: left;
        width: 100%;
        max-width: $container__size;
        position: relative;
        bottom: 30px;

        @media screen and (max-width: $small__tablet) {
            left: 20px;
            position: absolute;
            top: -40px;
        }

        @media screen and (max-width: $mobile) {
            top: -60px;
            left: 5px;
        }
    }

    &__img-left {
        align-self: flex-end;

        &::before {
            content: url('../../../../assets/img/common/wwd-dots.svg');

            @media screen and (max-width: $tablet) {
                content: url('../../../../assets/img/pageHeader/pttrns-dots-small-down.svg');
                // left: -20px;
                bottom: -30px;
                left: -30px;
                position: relative;
            }

            @media screen and (max-width: $large__mobile) {
                content: url('../../../../assets/img/pageHeader/pttrns-dots-mob-down.svg');

                top: 30px;
                left: -35px;
            }

            @media screen and (max-width: $mobile) {
                content: url('../../../../assets/img/pageHeader/pttrns-dots-xs-down.svg');
            }
        }
    }

    &__img-right {
        align-self: flex-start;
        justify-content: flex-end;
        top: -50px;
        position: relative;

        &::after {
            content: url('../../../../assets/img/dots-large.svg');
            position: relative;

            @media screen and (max-width: $tablet) {
                top: -20px;
                content: url('../../../../assets/img/pageHeader/pttrns-dots-small-up.svg');
            }

            @media screen and (max-width: $small__tablet) {
                top: -30px;
                right: -20px;
            }

            @media screen and (max-width: $large__mobile) {
                content: url('../../../../assets/img/pageHeader/pttrns-dots-mob-up.svg');
                right: 22px;
                top: -35px;
            }

            @media screen and (max-width: $mobile) {
                content: url('../../../../assets/img/pageHeader/pttrns-dots-xs-up.svg');
            }
        }
    }

    &__button-wrapper {
        margin-top: 70px;
        display: flex;
        align-items: center;

        &--moved {
            @media screen and (max-width: $mobile) {
                margin-left: 0;
                position: relative;
                right: 30px;
            }
        }

        @media screen and (max-width: $tablet) {
            margin-left: 0;
        }

        @media screen and (max-width: $mobile) {
            margin-left: 0;
            position: relative;
            // right: 30px;
            // margin-left: 20px;
        }

        &::before {
            display: block;
            content: '';
            width: 100%;
            max-width: 120px;
            height: 1px;
            background-color: $secondary;
            margin-right: 20px;
        }
        &::after {
            margin-left: 10px;
            display: block;
            content: '';
            width: 100%;
            max-width: 465px;
            height: 1px;
            background-color: $secondary;

            @media screen and (max-width: $desktop) {
                max-width: 300px;
            }
        }
    }
}
