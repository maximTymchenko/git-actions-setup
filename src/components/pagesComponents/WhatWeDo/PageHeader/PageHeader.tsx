import React, { FC } from 'react'
import './PageHeader.scss'
import Filter from '../../../common/Filter'
import Button from '../../../common/Button'
import { Link } from 'react-scroll'
import arrowDown from '../../../../assets/img/arr-down.png'
import wwdDots from '../../../../assets/img/common/wwd-dots.svg'
import dotsL from '../../../../assets/img/dots-large.svg'

interface IPageHeader {
  title: string
  subtitle: string
  id: string
  bgType: string
  offset: string
}

const PageHeader: FC = (props: IPageHeader) => {
  return (
    <div data-testid="page-header" className="page-header">
      {/* <Filter /> */}
      <div
        className={`page-header__wrapper page-header__wrapper--${props.bgType}`}
      >
        <div className="page-header__container">
          <div className="page-header__img-left" />
          <div className="page-header__title">
            <h2 className="title title--small">{props.subtitle}</h2>
            <h1 className="title title--section title--brackets">
              {props.title}
            </h1>
          </div>
          <div className="page-header__img-right" />
        </div>
        <div className="page-header__button-wrapper">
          <Link to={props.id} smooth={true} offset={props.offset}>
            <Button icon={arrowDown} type="primary" desc="Show how">
              Show how
            </Button>
          </Link>
        </div>
      </div>
    </div>
  )
}

export default PageHeader
