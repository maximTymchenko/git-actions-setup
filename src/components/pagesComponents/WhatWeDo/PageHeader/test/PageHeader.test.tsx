import React from 'react';
import PageHeader from '../PageHeader';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(
        <PageHeader title="Whar We Do" subtitle="Discover"></PageHeader>,
    );
    expect(getByTestId('page-header')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer
        .create(<PageHeader title="Whar We Do" subtitle="Discover"></PageHeader>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
