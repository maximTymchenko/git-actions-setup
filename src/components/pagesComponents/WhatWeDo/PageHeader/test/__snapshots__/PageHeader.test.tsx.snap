// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="page-header"
  data-testid="page-header"
>
  <div
    className="page-header__wrapper page-header__wrapper--undefined"
  >
    <div
      className="page-header__container"
    >
      <div
        className="page-header__img-left"
      />
      <div
        className="page-header__title"
      >
        <h2
          className="title title--small"
        >
          Discover
        </h2>
        <h1
          className="title title--section title--brackets"
        >
          Whar We Do
        </h1>
      </div>
      <div
        className="page-header__img-right"
      />
    </div>
    <div
      className="page-header__button-wrapper page-header__button-wrapper--undefined"
    >
      <a
        onClick={[Function]}
      >
        <button
          className="button button--primary button--undefined button--undefined"
          data-testid="button"
        >
          Show how
          <img
            alt="Show how"
            className="button__icon button__icon--undefined"
            data-testid="button__icon"
            src="test-file-stub"
          />
        </button>
      </a>
    </div>
  </div>
</div>
`;
